src/lib.rs

```diff
@@ -132,23 +142,47 @@ impl Context {
         }
     }
 
-    pub fn start_auth_session(&mut self, bind: ESYS_TR) -> Result<ESYS_TR> {
-        let sym_def = utils::TpmtSymDefBuilder::aes_256_cfb();
+    // TODO: Fix when compacting the arguments into a struct
+    #[allow(clippy::too_many_arguments)]
+    pub fn start_auth_session(
+        &mut self,
+        non_auth_sessions: (ESYS_TR, ESYS_TR, ESYS_TR),
+        tpm_key: ESYS_TR,
+        bind: ESYS_TR,
+        nonce: &[u8],
+        session_type: TPM2_SE,
+        symmetric: TPMT_SYM_DEF,
+        auth_hash: TPMI_ALG_HASH,
+    ) -> Result<ESYS_TR> {
+        if nonce.len() > 64 {
+            return Err(Tss2ResponseCode::new(TPM2_RC_SIZE));
```

Comment:
Should we define our own errors for this layer and abstraction layers above as the `Tss2ResponseCode` should IMO only come from the TPM commands sent from the TPM?

Reply:
(As in, only used when returning error codes from ESYS functions)

Reply:
I thought about that, but you'd need to occupy a block of response codes from the TSS2 space - again, I think I'll raise an issue to look into it and how to document it.

Reply:
Yes we should raise an issue about that and discuss it there!